### ISSUE TITLE
No way to access google.maps object

### ISSUE THREAD

**Issue by NathanAP** [NathanAP/vue-google-maps-community-fork#39]
Hi Guys,

Thanks so much for maintaining this folk.

A quick question on how to set more advanced options, e.g. `mapTypeControlOptions`

I cannot seem to access the actual map property

Here are the docs: 
https://developers.google.com/maps/documentation/javascript/controls#maps_control_positioning-typescript

Here is a related thread for react:
https://github.com/google-map-react/google-map-react/issues/6

Any guidance on how I can reposition mapTypeControl would be appreciated.

_Originally posted by @andyslack in https://github.com/NathanAP/vue-google-maps-community-fork/discussions/36_

---------------

As you can see in the refered discussion, Andy and I talked about we access the `google.maps` to use some of its feature in `Map options` props, for example. I'm almost sure we could add to `index.html` a reference to Google Maps, but it would break other things and this is not a good solution.

Could we export `google.maps` directly from the library? Is that a good solution?

**Comment by NathanAP** [comment]
Can we just export `google.maps` object in [main.js](https://github.com/NathanAP/vue-google-maps-community-fork/blob/main/src/main.js)? 🤔

**Comment by metinjakupi** [comment]
Im using like this  const google = await mapRef.value.$gmapApiPromiseLazy()

**Comment by NathanAP** [comment]
> Im using like this const google = await mapRef.value.$gmapApiPromiseLazy()

oh, it worked... thats explain a lot the docs example, I knew that I was doing something wrong.
Then what should we do about #42 ?

**Comment by metinjakupi** [comment]
#42 I don't think this is good approach because setTimeout is used (my opinion)

**Comment by NathanAP** [comment]
I agree. I used setTimeout as an example. If you need to wait for something else inside your screen, it shouldn't be a problem.

The main problem(?) I see here is that we need a GMap ref. Do you consider this a problem?

@andyslack, can you give us your opinion as well?

**Comment by NathanAP** [comment]
Ok, I'll remove my PR and add more examples to the docs in the next update we do there. Thanks for the help guys! I appreciate your time with us.

**Comment by johnwc** [comment]
Just an FYI, I have a local version that I took from this repo and ported it to TypeScript. That's what we currently use, it also has some minor improvements for the $emit events to include, as event parameters, the InfoWindow and Marker objects that triggered the event. Once I was happy with everything, I was going to push it to github and npmjs as a library.

**Comment by NathanAP** [comment]
That's great to hear! We were discussing about turning this library into API composition but we didn't had the time for it. We probably would turn into typescript eventually as well. 

If you want to make a big merge and make an unique library we can do it together, let me know if it's your intention or not :) all I did so far is keep the community in touch with it and fix some important bugs, so I'm okay with every help and improvement in the code.

**Comment by andyslack** [comment]
I think moving to Typescript would be a fantastic upgrade! 

I also feel having access to the `google.maps` without having to use `this.` would be preferable but as long as the documentation is good, it will be fine. 

Perhaps we can host working examples of each `example` on codesandbox.io or StackBlitz

**Comment by NathanAP** [comment]
TBH I have zero knowledge in Typescript, I have some projects with Vue but all of them are made in Javascript. If someone wants to do it, I'm 100% available to help in the best way possible.

**Comment by NathanAP** [comment]
@johnwc do you mind send me a message when you update your version to npm?

**Comment by johnwc** [comment]
@NathanAP Sure will

**Comment by NathanAP** [comment]
Closing this since #42 was closed as well. You guys can keep the conversation up if you want to.
